src/sidebar/store/test/use-store-test.js

```diff
@@ -16,90 +18,262 @@ const reducer = (state = initialState, action) => {
   }
 };
 
-describe('useStore', () => {
-  let renderCount;
-  let testStore;
-  let TestComponent;
-
-  beforeEach(() => {
-    renderCount = 0;
-
-    // eslint-disable-next-line react/display-name
-    TestComponent = () => {
-      renderCount += 1;
-      const aValue = useStore(store => store.getState().value);
-      return <div>{aValue}</div>;
-    };
-
-    testStore = createStore(reducer);
-    $imports.$mock({
-      '../util/service-context': {
-        useService: name => (name === 'store' ? testStore : null),
-      },
-    });
-  });
+// Store module for use with `createStore` in tests.
+const thingsModule = {
+  namespace: 'things',
+
+  init: () => ({
+    things: [],
+  }),
+
+  update: {
+    ADD_THING(state, action) {
+      if (state.things.some(t => t.id === action.thing.id)) {
+        return {};
+      }
+      return { things: [...state.things, action.thing] };
+    },
+  },
+
+  actions: {
+    addThing(id) {
+      return { type: 'ADD_THING', thing: { id } };
+    },
+  },
+
+  selectors: {
+    thingCount(state) {
+      return state.things.length;
+    },
+
+    getThing(state, id) {
+      return state.things.find(t => t.id === id);
+    },
+  },
+};
 
+describe('sidebar/store/use-store', () => {
   afterEach(() => {
     $imports.$restore();
   });
 
-  it('returns result of `callback(store)`', () => {
-    const wrapper = mount(<TestComponent />);
-    assert.equal(wrapper.text(), '10');
-  });
+  // Tests for deprecated `useStore` function.
```

Comment:
The diff in the `describe('useStore', ...` block are just whitespace changes. See [the diff without whitespace changes](https://github.com/hypothesis/client/pull/2800/files?diff=unified&w=1).